### ISSUE TITLE
bot crashes with repeated non-ascii characters

### ISSUE THREAD

**Issue by sijis** [errbotio/errbot#455]
Reported in G+ by D. Hynes

Here is the error I see in the log file when the bot crashes: http://pastebin.com/y8S17srC

I have been able to successfully duplicate this now, I login to the server via an actual telnet client to remove a lot of the processing that occurs in actual IRC clients and send a private message to the bot containing the characters ï¿½ over and over and over again (easily a few hundred) in the message and then the error hits the log and the bot promptly dies.﻿

He tested this against Text backed and IRC.

**Comment by sijis** [comment]
For full reference, the traceback is

```
Traceback (most recent call last):
  File "/usr/bin/err.py", line 261, in <module>
    main(backend, logger, config, restore)
  File "/usr/lib/python2.7/site-packages/errbot/main.py", line 90, in main
    bot.serve_forever()
  File "/usr/lib/python2.7/site-packages/errbot/backends/irc.py", line 462, in serve_forever
    self.conn.start()
  File "/usr/lib/python2.7/site-packages/irc/bot.py", line 265, in start
    super(SingleServerIRCBot, self).start()
  File "/usr/lib/python2.7/site-packages/irc/client.py", line 1247, in start
    self.reactor.process_forever()
  File "/usr/lib/python2.7/site-packages/irc/client.py", line 278, in process_forever
    self.process_once(timeout)
  File "/usr/lib/python2.7/site-packages/irc/client.py", line 259, in process_once
    self.process_data(i)
  File "/usr/lib/python2.7/site-packages/irc/client.py", line 216, in process_data
    c.process_data()
  File "/usr/lib/python2.7/site-packages/irc/client.py", line 583, in process_data
    for line in self.buffer:
  File "/usr/lib/python2.7/site-packages/irc/buffer.py", line 102, in lines
    self.handle_exception()
  File "/usr/lib/python2.7/site-packages/irc/buffer.py", line 100, in lines
    yield line.decode(self.encoding, self.errors)
  File "/usr/lib64/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xc3 in position 509: unexpected end of data
```

**Comment by dword4** [comment]
Should also mention that the character never shows up in the logs even when set to debug level, so its breaking before the logging portion of the code, the only way I could actually see the characters was doing a packet capture on the system the bot was running on.

**Comment by gbin** [comment]
I might have a clue about what is going on...

**Comment by zoni** [comment]
From looking at that backtrace, I gather this is an issue with the `irc` package rather than Err itself.

IRC is [opinionated](https://bitbucket.org/jaraco/irc/issues/40/unicodedecodeerror) about the data it receives.

I suppose we could switch the default buffer to [something more forgiving](https://bitbucket.org/jaraco/irc/src/57f6380dd2a547b09b0eaf7a63d497ed2f68b62b/README.txt?at=default&fileviewer=file-view-default#README.txt-143:204). I'd be in favor of using `irc.client.ServerConnection.buffer_class.errors = 'replace'`.

**Comment by gbin** [comment]
Should be fixed by 45299a0498d9d0fdcfe86c3326a71acf16177442 please reopen if it is not
